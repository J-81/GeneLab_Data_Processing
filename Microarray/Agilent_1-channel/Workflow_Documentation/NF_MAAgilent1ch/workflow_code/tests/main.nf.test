nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Direct runsheet version: SKIPVV : Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                runsheetPath = "../examples/runsheet/GLDS-367_microarray_v0_runsheet.csv"
                gldsAccession = "GLDS-367"
                biomart_attribute = "agilent_wholegenome_4x44k_v1"
                limit_biomart_query = 300
                skipVV = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${params.outputDir}/${params.gldsAccession}").list(),
                            ).match()
        }

    }
    test("GLDS-367: SKIPVV : Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-367"
                gldsAccession = "GLDS-367"
                biomart_attribute = "agilent_wholegenome_4x44k_v1"
                limit_biomart_query = 300
                skipVV = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${params.outputDir}/${params.gldsAccession}").list(),
                            ).match()
        }

    }

}
