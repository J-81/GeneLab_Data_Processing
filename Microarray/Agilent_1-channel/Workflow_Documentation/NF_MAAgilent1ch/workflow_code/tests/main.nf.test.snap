{
    "Direct runsheet version: SKIPVV : Should run without failures": {
        "content": [
            [
                "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                "differential_expression.csv:md5,7aabea95a4b1d09ec26fb826daa17f08",
                "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                "visualization_output_table.csv:md5,6cac37993daf664d1be489b1f5a9befd"
            ],
            [
                "GLDS-367_microarray_v0_runsheet.csv:md5,deeb1c31bfda3f10627be9842423491d"
            ]
        ],
        "timestamp": "2023-01-26T23:10:39+0000"
    },
    "GLDS-367: SKIPVV : Should run without failures": {
        "content": [
            [
                "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                "differential_expression.csv:md5,7aabea95a4b1d09ec26fb826daa17f08",
                "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                "visualization_output_table.csv:md5,6cac37993daf664d1be489b1f5a9befd"
            ],
            [
                "GLDS-367_metadata_GLDS-367-ISA.zip:md5,130bb4f9fded2750631cfed98c173c06",
                "GLDS-367_microarray_v0_runsheet.csv:md5,deeb1c31bfda3f10627be9842423491d"
            ]
        ],
        "timestamp": "2023-01-26T23:10:40+0000"
    }
}